src/ui/simulator/windows/options/optimization/optimization.cpp

```diff
@@ -729,6 +734,75 @@ void Optimization::onPopupMenuTransmissionCapacities(Component::Button&, wxMenu&
                  this);
 }
 
+void Optimization::onPopupMenuExportMPSstatus(Component::Button&, wxMenu& menu, void*)
+{
+    // Create the item for no mps export
+    {
+        const Data::mpsExportStatus& value = Data::mpsExportStatus::NO_EXPORT;
+        wxMenuItem* it = Menu::CreateItem(
+                        &menu,
+                        wxID_ANY,
+                        mpsExportStatusToString(value), 
+                        mpsExportIcon(value), 
+                        wxEmptyString);
+        menu.Connect(it->GetId(),
+            wxEVT_COMMAND_MENU_SELECTED,
+            wxCommandEventHandler(Optimization::onSelectNoMPSexport),
+            nullptr,
+            this);
+    }
+
+    // Create the item to export mps associated to first optimization
+    {
+        const Data::mpsExportStatus& value = Data::mpsExportStatus::EXPORT_FIRST_OPIM;
+        wxMenuItem* it = Menu::CreateItem(
+                        &menu,
+                        wxID_ANY,
+                        mpsExportStatusToString(value),
+                        mpsExportIcon(value),
+                        wxEmptyString);
+        menu.Connect(it->GetId(),
+            wxEVT_COMMAND_MENU_SELECTED,
+            wxCommandEventHandler(Optimization::onSelectExportMPSfirstOptimization),
+            nullptr,
+            this);
+    }
+
+    // Create the item to export mps associated to second optimization
+    {
+        const Data::mpsExportStatus& value = Data::mpsExportStatus::EXPORT_SECOND_OPIM;
+        wxMenuItem* it = Menu::CreateItem(
+                        &menu,
+                        wxID_ANY,
+                        mpsExportStatusToString(value),
+                        mpsExportIcon(value),
+                        wxEmptyString);
+        menu.Connect(it->GetId(),
+            wxEVT_COMMAND_MENU_SELECTED,
+            wxCommandEventHandler(Optimization::onSelectExportMPSsecondOptimization),
+            nullptr,
+            this);
+    }
+
+    // Create the item to export mps associated to both optimizations
+    {
+        const Data::mpsExportStatus& value = Data::mpsExportStatus::EXPORT_BOTH_OPTIMS;
+        wxMenuItem* it = Menu::CreateItem(
+                        &menu,
+                        wxID_ANY,
+                        mpsExportStatusToString(value),
+                        mpsExportIcon(value),
+                        wxEmptyString);
+        menu.Connect(it->GetId(),
+            wxEVT_COMMAND_MENU_SELECTED,
+            wxCommandEventHandler(Optimization::onSelectExportMPSbothOptimizations),
+            nullptr,
+            this);
+    }
```

Comment:
for loop ? Like here https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/832

Reply:
I did not introduce a loop here, but a template function makes thing more clear and avoids duplicated code